src/storage/mongodb/SettingStorage.ts

```diff
@@ -304,6 +304,50 @@ export default class SettingStorage {
     }
   }
 
+  public static async saveCryptoSettings(tenantID: string, cryptoSettingToSave: KeySetting): Promise<void> {
+    // Build internal structure
+    const settingsToSave = {
+      id: cryptoSettingToSave.id,
+      identifier: 'crypto',
+      lastChangedOn: new Date(),
+      content: {
+        crypto: cryptoSettingToSave.crypto
+      },
+    } as SettingDB;
+    // Save
+    await this.saveSettings(tenantID, settingsToSave);
+  }
+
+  public static async getCryptoSettings(tenantID: string): Promise<KeySetting> {
+    // Get the Crypto Key settings
+    const settings = await SettingStorage.getSettings(tenantID,
+      { identifier: TenantComponents.CRYPTO },
+      Constants.DB_PARAMS_MAX_LIMIT);
+    if (settings.count > 0) {
+      const cryptoSetting = {
+        formerKey: settings.result[0].content.crypto.formerKey,
```

Comment:
You should test for the formerKey attribute existence in the result before assignin: 

`settings.result[0].content.crypto.formerKey ? settings.result[0].content.crypto.formerKey : null`

And do the same for other formerKey related settings.